packages/gatsby-recipes/src/renderer/render.js

```diff
@@ -253,48 +257,53 @@ const render = (recipe, cb, inputs = {}, isApply, isStream, name) => {
   const renderResources = isDrained => {
     result = RecipesReconciler.render(recipeWithWrapper, plan, name)
 
-    const resources = transformToPlan(result)
+    const resourcesArray = transformToPlan(result)
 
     const isDone = () => {
+      let result
       // Mostly for validation stage that checks that there's no resources
       // in the initial step — this done condition says no resources were found
       // and there's no inflight resource work (resources will be empty until the
       // first resource returns).
       //
       // We use "inFlightCache" because the queue doesn't immediately show up
       // as having things in it.
-      if (resources.length === 0 && ![...inFlightCache.values()].some(a => a)) {
-        return true
+      if (
+        resourcesArray.length === 0 &&
+        ![...inFlightCache.values()].some(a => a)
```

Comment:
Old code but: I don't know how big htis cache can grow. If it's big enough wouldn't a boolean flag (on the same level as `inFlightCache`) be far more efficient than having to do all of this? It's also done below.

Nit: If nothing else, `![].some` is the same as `[].every`, though the choice usually depends on which case is more likely to return early. Not sure here.

Reply:
a single recipe should almost never have more than a few 100 items in this inflightcache — I measured how long this code took to run and it was very small % of the total time so left it out of this refactor.